supervisor/api/utils.py

```diff
@@ -101,13 +101,15 @@ async def wrap_api(api, *args, **kwargs):
             """Return api information."""
             try:
                 msg_data = await method(api, *args, **kwargs)
+                if isinstance(msg_data, (web.Response, web.StreamResponse)):
+                    return msg_data
                 msg_type = content
-            except (APIError, APIForbidden) as err:
+            except (APIError, APIForbidden, HassioError) as err:
                 msg_data = str(err).encode()
-                msg_type = CONTENT_TYPE_BINARY
+                msg_type = error_type or CONTENT_TYPE_BINARY
             except HassioError:
                 msg_data = b""
-                msg_type = CONTENT_TYPE_BINARY
+                msg_type = error_type or CONTENT_TYPE_BINARY
```

Comment:
```suggestion
            except HassioError as err:
                msg_data = str(err).encode()
                msg_type = CONTENT_TYPE_BINARY
                msg_type = error_type or CONTENT_TYPE_BINARY
```
Tbh, I'm not entirely sure how this code passed the linter since `HassioError` is in there twice in two different `except` clauses. But `APIForbidden` extends from `APIError` which extends from `HassioError`. So if you have `HassioError` in the first one then the others are just redundant. And the second except for just `HassioError` will never be called.

Although really I think the code here should be this:
```py
            status = 200
            try:
                msg_data = await method(api, *args, **kwargs)
                if isinstance(msg_data, (web.Response, web.StreamResponse)):
                    return msg_data
                msg_type = content
            except APIError as err:
                msg_data = str(err).encode()
                msg_type = error_type or CONTENT_TYPE_BINARY
                status = err.status
            except HassioError:
                msg_data = str(err).encode()
                msg_type = error_type or CONTENT_TYPE_BINARY
                status = 400

            return web.Response(body=msg_data, content_type=msg_type, status=status)
```
Because right now we don't reply with 403s when an `APIForbidden` is raised. But I don't think you're actually using this annotation at all anymore are you? Perhaps we should just leave all this for another PR.

Reply:
Ok actually start from the last comment in these 3. Since if you make that change, these other two are resolved. Sorry I realized after the other two that there might be a better way to do this then a new annotation.

Reply:
TBH this was actually committed accidentally - it just slipped in because I have a habit of running `git add -u ; pre-commit ; git diff` to re-format the code and forgot to revert these changes when I implemented `api_process_custom` :facepalm: